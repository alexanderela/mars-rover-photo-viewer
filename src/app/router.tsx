import {
  createBrowserRouter,
  redirect,
} from "react-router-dom";
import { EnlargedPhotoViewer } from "../features/EnlargedPhotoViewer";

export const routesConfig = [
  {
    path: "/",
    lazy: async () => {
      const { default: App } = await import("./App");

      return {
        element: (
          <App />
        ),
      }
    },
    children: [
      {
        index: true,
        loader: async () => redirect("rovers/curiosity?page=1")
      },
      {
        path: "rovers/:rover",
        lazy: async () => {
          const { PhotoViewer } = await import("../features/photos/PhotoViewer");
    
          return {
            element: (
              <PhotoViewer />
            )
          }
        },
        children: [
          {
            path: "id/:id",
            element: <EnlargedPhotoViewer />
          }
        ]
      },
    ]
  },
];

export const router = createBrowserRouter(routesConfig);