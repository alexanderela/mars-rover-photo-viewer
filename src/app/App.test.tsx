import { describe, expect, it } from "vitest";
import { render, waitFor } from "../tests/test-utils";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { routesConfig } from "./router";

describe("<App />", () => {
  it("renders App component", async () => {
    const router = createMemoryRouter(routesConfig, {
      initialEntries: ["/", "/photos/rovers/curiosity?page=1"],
      initialIndex: 1
    });
    const result = render(<RouterProvider router={router} />);

    await waitFor(() => {
      const element = result.getByTestId(/App/i);
      expect(element).toBeInTheDocument();
      expect(element).toMatchSnapshot();
    });
  });
});