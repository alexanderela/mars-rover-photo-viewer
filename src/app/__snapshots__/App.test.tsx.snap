// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > renders App component 1`] = `
<div
  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1o5nixv-MuiGrid2-root"
  data-testid="App"
>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1obpxxd-MuiGrid2-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-qtfspc-MuiTypography-root"
    >
      Mars Rover Photo Viewer
    </h5>
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1ssg4ni-MuiAvatar-root"
    >
      U
    </div>
  </div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-1 css-pvedf0-MuiGrid2-root"
    data-testid="SideNav"
  />
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-11 css-qvo7h4-MuiGrid2-root"
    data-testid="Content"
  >
    <div
      class="MuiBox-root css-1lekzkb"
    >
      <div
        class="MuiTabs-root css-mz7c2a-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-1wxkzlj-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Curiosity
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Opportunity
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Spirit
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1tb7sr5-MuiTabs-indicator"
            style="left: 0px; width: 0px; display: none;"
          />
        </div>
      </div>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-text css-19zxguc-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-6qpcc8-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-6qpcc8-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1sp5kfn-MuiCircularProgress-root"
      data-testid="CircularProgressLoader"
      role="progressbar"
      style="width: 60px; height: 60px;"
    >
      <svg
        class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`<App /> > renders App component 2`] = `
<div
  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1o5nixv-MuiGrid2-root"
  data-testid="App"
>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1obpxxd-MuiGrid2-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-qtfspc-MuiTypography-root"
    >
      Mars Rover Photo Viewer
    </h5>
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1ssg4ni-MuiAvatar-root"
    >
      U
    </div>
  </div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-1 css-pvedf0-MuiGrid2-root"
    data-testid="SideNav"
  />
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-11 css-qvo7h4-MuiGrid2-root"
    data-testid="Content"
  >
    <div
      class="MuiBox-root css-1lekzkb"
    >
      <div
        class="MuiTabs-root css-mz7c2a-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-1wxkzlj-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Curiosity
              <span
                class="MuiTabs-indicator css-1tb7sr5-MuiTabs-indicator"
                style="display: none;"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Opportunity
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Spirit
            </button>
          </div>
        </div>
      </div>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-text css-19zxguc-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-6qpcc8-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-6qpcc8-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1sp5kfn-MuiCircularProgress-root"
      data-testid="CircularProgressLoader"
      role="progressbar"
      style="width: 60px; height: 60px;"
    >
      <svg
        class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`<App /> > renders App component 3`] = `
<div
  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1o5nixv-MuiGrid2-root"
  data-testid="App"
>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1obpxxd-MuiGrid2-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-qtfspc-MuiTypography-root"
    >
      Mars Rover Photo Viewer
    </h5>
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1ssg4ni-MuiAvatar-root"
    >
      U
    </div>
  </div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-1 css-pvedf0-MuiGrid2-root"
    data-testid="SideNav"
  />
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-11 css-qvo7h4-MuiGrid2-root"
    data-testid="Content"
  >
    <div
      class="MuiBox-root css-1lekzkb"
    >
      <div
        class="MuiTabs-root css-mz7c2a-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-s2t35c-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-1wxkzlj-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Curiosity
              <span
                class="MuiTabs-indicator css-1tb7sr5-MuiTabs-indicator"
                style="display: none;"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Opportunity
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1sx2hv0-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Spirit
            </button>
          </div>
        </div>
      </div>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-text css-19zxguc-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-6qpcc8-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-6qpcc8-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1sp5kfn-MuiCircularProgress-root"
      data-testid="CircularProgressLoader"
      role="progressbar"
      style="width: 60px; height: 60px;"
    >
      <svg
        class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
